src/rapidata/service/credential_manager.py

```diff
@@ -0,0 +1,227 @@
+                return None
+
+            data = response.json()
+            return BridgeToken(read_key=data["readKey"], write_key=data["writeKey"])
+        except requests.RequestException as e:
+            print(f"Failed to get bridge tokens: {e}")
+            return None
+
+    def _poll_read_key(self, read_key: str) -> Optional[str]:
+        """Poll the read key endpoint until we get an access token."""
+        read_endpoint = f"{self.endpoint}/Identity/ReadBridgeToken"
+        start_time = time.time()
+
+        while time.time() - start_time < self.poll_timeout:
+            try:
+                response = requests.get(
+                    read_endpoint, params={"readKey": read_key}, verify=self.cert_path
+                )
+
+                if response.status_code == 200:
+                    return response.json().get("accessToken")
+                elif response.status_code == 202:
+                    # Still processing
+                    time.sleep(self.poll_interval)
+                    continue
+                else:
+                    # Error occurred
+                    print(f"Error polling read key: {response.status_code}")
+                    return None
+
+            except requests.RequestException as e:
+                print(f"Error polling read key: {e}")
+                return None
+
+        print("Polling timed out")
+        return None
+
+    def _create_client(self, access_token: str) -> Optional[Tuple[str, str, str]]:
+        """Create a new client using the access token."""
+        try:
+            # set the display name to the hostname
+            display_name = f"{os.uname().nodename} - CLI"
+            response = requests.post(
+                f"{self.endpoint}/Client",
+                headers={
+                    "Authorization": f"Bearer {access_token}",
+                    "Content-Type": "application/json",
+                    "Accept": "*/*",
+                },
+                json={"displayName": display_name},
+                verify=self.cert_path,
+            )
+            response.raise_for_status()
+            data = response.json()
+            return data.get("clientId"), data.get("clientSecret"), display_name
+        except requests.RequestException as e:
+            print(f"Failed to create client: {e}")
+            return None
+
+    def _create_new_credentials(self) -> Optional[ClientCredential]:
+        bridge_endpoint = self._get_bridge_tokens()
+        if not bridge_endpoint:
+            return None
+
+        auth_url = f"{self.endpoint}/connect/authorize/external?clientId=rapidata-cli&scope=openid profile email&writeKey={bridge_endpoint.write_key}"
+        webbrowser.open(auth_url)
+
+        access_token = self._poll_read_key(bridge_endpoint.read_key)
+        if not access_token:
+            return None
+
+        client_id, client_secret, display_name = self._create_client(access_token)
```

Comment:
If `_create_client()` returns None this will throw an error, as None can't be unpacked into the three values